src/disk_io_thread_pool.cpp

```diff
@@ -70,8 +70,6 @@ namespace libtorrent {
 	void disk_io_thread_pool::abort(bool wait)
 	{
 		std::unique_lock<std::mutex> l(m_mutex);
-		if (m_abort) return;
-		m_max_threads = 0;
```

Comment:
in this patch, `m_max_threads` is used to start up threads and to configure the number of threads once started up. It's not used to synchronize the shutdown, as that's the problem, you can't tell the difference between not having started up yet and wanting to shut down. `m_abort` is used to signal that we want to shut down, not setting the number of threads to 0